azurerm/internal/services/consumption/registration.go

```diff
@@ -0,0 +1,37 @@
+package consumption
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	consumptionBudgetResourceGroupName = "azurerm_consumption_budget_resource_group"
```

Comment:
It feels weird to use this format but I can see the value after thinking it through. Do you mind leaving a comment about why we're using this format in case we come back and forget why it's done this way

Reply:
Clarification added in https://github.com/terraform-providers/terraform-provider-azurerm/pull/9201/commits/2a6d556ce4ad76cca668bef55d9c148373a921d6